sql/sql_class.h

```diff
@@ -3991,6 +3995,13 @@ class THD : public MDL_context_owner,
   */
   void claim_memory_ownership();
 
+  bool is_thd_priority_alt() { return thd_priority_alt; }
+  void mark_thd_priority_as_alt() {
+    mysql_mutex_lock(&LOCK_thd_data);
```

Comment:
May be make `thd_priority_alt` an instance of `std::atomic_bool` to avoid mutex locks?